modules/internal/String.chpl

```diff
@@ -1587,6 +1587,32 @@ module String {
       return _join(S);
     }
 
+    /*
+      :args listToJoin: list contaning elements whose string values will be concatenated
```

Comment:
For the sake of consistency, it might be good if the docstring for this method mirrors the docstrings of the `join` overloads for tuples and arrays. In example, something like...

```
    /*
      Same as the varargs version, but with all the strings in a list.

      .. code-block:: chapel
          var someList = new list(string);
          someList.append("a");
          someList.append("10");
          someList.append("d");
          var x = "|".join(someList);
          writeln(x); // prints: "a|10|d"
     */
```

You did a good thing there by listing the argument (frankly the rest of the join methods should list their arguments as well, but that can wait). 

Also make sure to list the return type!